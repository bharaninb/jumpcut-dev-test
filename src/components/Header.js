import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import PropTypes from 'prop-types';
import React from 'react';

export const HeaderComponent = ({ title }) => (
  <AppBar position="absolute">
    <Toolbar>
      <Typography component="h1" variant="h6" color="inherit" noWrap>
        {title}
      </Typography>
    </Toolbar>
  </AppBar>
);

HeaderComponent.propTypes = {
  title: PropTypes.string.isRequired,
};
