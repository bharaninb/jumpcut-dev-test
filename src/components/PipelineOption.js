import FormControl from '@material-ui/core/FormControl';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import FormLabel from '@material-ui/core/FormLabel';
import Radio from '@material-ui/core/Radio';
import RadioGroup from '@material-ui/core/RadioGroup';
import { withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import React from 'react';

const styles = theme => ({
  root: {
    display: 'flex',
  },
  formControl: {
    margin: theme.spacing.unit * 3,
  },
  group: {
    margin: `${theme.spacing.unit}px 0`,
  },
});

class PipelineOption extends React.Component {
  handleChange = (event) => {
    const { onChange } = this.props;
    onChange(event.target.value);
  };

  render() {
    const { classes, pipelineOption } = this.props;

    return (
      <div className={classes.root}>
        <FormControl component="fieldset" className={classes.formControl}>
          <FormLabel component="legend">PipelineOption</FormLabel>
          <RadioGroup
            aria-label="PipelineOption"
            name="PipelineOption"
            className={classes.group}
            value={pipelineOption}
            onChange={this.handleChange}
          >
            <FormControlLabel value="none" control={<Radio />} label="None" />
            <FormControlLabel value="accumulator" control={<Radio />} label="Accumulator" />
            <FormControlLabel value="isEven" control={<Radio />} label="IsEven" />
          </RadioGroup>
        </FormControl>
      </div>
    );
  }
}

PipelineOption.propTypes = {
  classes: PropTypes.shape().isRequired,
  pipelineOption: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};

export const PipelineOptionComponent = withStyles(styles)(PipelineOption);
