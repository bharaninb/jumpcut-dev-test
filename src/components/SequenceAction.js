import Button from '@material-ui/core/Button';
import { withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import React from 'react';

const styles = theme => ({
  root: {
    padding: theme.spacing.unit * 3,
    textAlign: 'center',
  },
});

class SequenceAction extends React.Component {
  handleNext = () => {
    const { handleNext, appSequence } = this.props;
    handleNext(appSequence.sequenceGen);
  };

  handleActive = () => {
    const { activateSequence, appSequence } = this.props;
    activateSequence(appSequence.selectedSequence, appSequence.pipelineOption);
  };

  render() {
    const { classes, appSequence } = this.props;

    if (appSequence.isActive) {
      return (
        <div className={classes.root}>
          <Button variant="contained" color="primary" size="large" onClick={this.handleNext} id="nextBtn">
              Next
          </Button>
        </div>
      );
    }

    return (
      <div className={classes.root}>
        <Button variant="contained" color="primary" size="large" onClick={this.handleActive} id="activateBtn">
            Activate
        </Button>
      </div>
    );
  }
}

SequenceAction.propTypes = {
  classes: PropTypes.shape().isRequired,
  appSequence: PropTypes.shape().isRequired,
  activateSequence: PropTypes.func.isRequired,
  handleNext: PropTypes.func.isRequired,
};

export const SequenceActionComponent = withStyles(styles)(SequenceAction);
