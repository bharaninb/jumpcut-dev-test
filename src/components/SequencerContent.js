import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';
import { withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import React from 'react';
import { uid } from 'react-uid';

const styles = theme => ({
  root: {
    flexGrow: 1,
    padding: theme.spacing.unit,
  },
  paper: {
    padding: theme.spacing.unit * 2,
    textAlign: 'center',
    color: theme.palette.text.secondary,
  },
  button: {
    textAlign: 'center',
  },
});

const SequencerContent = ({ classes, sequenceValues }) => {
  const valueItems = sequenceValues.map((value, index) => (
    <Grid item key={uid(value, index)}>
      <Paper className={classes.paper}>{value}</Paper>
    </Grid>
  ));

  return (
    <div className={classes.root}>
      <Grid container spacing={24}>
        {valueItems}
      </Grid>
    </div>
  );
};

SequencerContent.propTypes = {
  classes: PropTypes.shape().isRequired,
  sequenceValues: PropTypes.arrayOf(PropTypes.number).isRequired,
};

export const SequencerContentComponent = withStyles(styles)(SequencerContent);
