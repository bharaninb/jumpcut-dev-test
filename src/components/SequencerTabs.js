import AppBar from '@material-ui/core/AppBar';
import { withStyles } from '@material-ui/core/styles';
import Tab from '@material-ui/core/Tab';
import Tabs from '@material-ui/core/Tabs';
import PropTypes from 'prop-types';
import React from 'react';

import { availableSequencers } from '../constants';

const styles = theme => ({
  root: {
    flexGrow: 1,
    width: '100%',
    backgroundColor: theme.palette.background.paper,
    marginTop: '65px',
  },
});

class SequencerTabs extends React.Component {
  handleChange = (event, value) => {
    const { onChange } = this.props;
    onChange(value);
  };

  render() {
    const { classes, selectedSequence } = this.props;

    const tabList = availableSequencers.map(sequence => (
      <Tab label={sequence.name} value={sequence.id} key={sequence.id} />
    ));

    return (
      <div className={classes.root}>
        <AppBar position="static" color="default">
          <Tabs
            id="sequenceTab"
            value={selectedSequence}
            onChange={this.handleChange}
            indicatorColor="primary"
            textColor="primary"
            scrollable
            scrollButtons="auto"
          >
            {tabList}
          </Tabs>
        </AppBar>
      </div>
    );
  }
}

SequencerTabs.propTypes = {
  classes: PropTypes.shape().isRequired,
  selectedSequence: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};

export const SequencerTabsComponent = withStyles(styles)(SequencerTabs);
