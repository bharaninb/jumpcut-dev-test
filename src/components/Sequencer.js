import PropTypes from 'prop-types';
import React from 'react';

import { SequencerContentComponent } from './SequencerContent';
import { SequencerTabsComponent } from './SequencerTabs';
import { PipelineOptionComponent } from './PipelineOption';
import { SequenceActionComponent } from './SequenceAction';

class SequencerComponent extends React.Component {
  render() {
    const {
      appSequence,
      selectSequence,
      fetchNextSequenceValue,
      selectPipeLine,
      activateSequence,
    } = this.props;

    return (
      <main>
        <SequencerTabsComponent
          selectedSequence={appSequence.selectedSequence}
          onChange={selectSequence}
        />
        <PipelineOptionComponent
          pipelineOption={appSequence.pipelineOption}
          onChange={selectPipeLine}
        />
        <SequencerContentComponent sequenceValues={appSequence.sequenceValues} />
        <SequenceActionComponent
          activateSequence={activateSequence}
          handleNext={fetchNextSequenceValue}
          appSequence={appSequence}
        />
      </main>
    );
  }
}

SequencerComponent.propTypes = {
  appSequence: PropTypes.shape().isRequired,
  selectSequence: PropTypes.func.isRequired,
  fetchNextSequenceValue: PropTypes.func.isRequired,
  selectPipeLine: PropTypes.func.isRequired,
  activateSequence: PropTypes.func.isRequired,
};

export { SequencerComponent };
