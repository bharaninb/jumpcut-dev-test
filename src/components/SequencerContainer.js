import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import {
  selectSequence,
  fetchNextSequenceValue,
  activateSequence,
  selectPipeLine,
} from '../redux/actions';
import { SequencerComponent } from './Sequencer';

const mapStateToProps = state => ({
  appSequence: state.appSequence,
});

const mapDispatchToProps = dispatch => ({
  selectSequence: bindActionCreators(selectSequence, dispatch),
  fetchNextSequenceValue: bindActionCreators(fetchNextSequenceValue, dispatch),
  selectPipeLine: bindActionCreators(selectPipeLine, dispatch),
  activateSequence: bindActionCreators(activateSequence, dispatch),
});

export const SequencerContainer = connect(
  mapStateToProps,
  mapDispatchToProps,
)(SequencerComponent);
