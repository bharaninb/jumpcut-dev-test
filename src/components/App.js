import React from 'react';

import { HeaderComponent } from './Header';
import { SequencerContainer } from './SequencerContainer';

export const AppComponent = () => (
  <React.Fragment>
    <HeaderComponent title="Jumpcut Dev Test" />
    <SequencerContainer />
  </React.Fragment>
);
