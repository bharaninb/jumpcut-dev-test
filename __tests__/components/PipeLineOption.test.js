import { shallow } from 'enzyme';
import React from 'react';

import { PipelineOptionComponent } from '../../src/components/PipelineOption';

describe('<PipelineOptionComponent />', () => {
  it('should match snapshot', () => {
    const component = shallow(
      <PipelineOptionComponent pipelineOption="none" onChange={jest.fn()} />,
    );
    expect(component).toMatchSnapshot();
  });
});
