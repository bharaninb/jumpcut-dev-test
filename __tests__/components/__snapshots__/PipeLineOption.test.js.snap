// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PipelineOptionComponent /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(PipelineOption)
    onChange={[MockFunction]}
    pipelineOption="none"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "formControl": "PipelineOption-formControl-2",
        "group": "PipelineOption-group-3",
        "root": "PipelineOption-root-1",
      },
      "onChange": [MockFunction],
      "pipelineOption": "none",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "formControl": "PipelineOption-formControl-2",
          "group": "PipelineOption-group-3",
          "root": "PipelineOption-root-1",
        },
        "onChange": [MockFunction],
        "pipelineOption": "none",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
